### DISCUSSION TITLE
Is there a way to use Volar with IntelliJ ?

### DISCUSSION THREAD

**Discussion by stephane303** [vuejs/language-tools#571] [Q&A] [answered]
I would like to avoid using 2 IDE

**Comment by johnsoncodehk** [comment]
We need LSP implement for WebStorm, fortunately it seems to exist: https://github.com/ballerina-platform/lsp4intellij
If someone wants to do it, you needs to make `@volar/vue-language-server` work with LSP4IntelliJ.
See https://github.com/johnsoncodehk/volar#communitys-language-client-implements, and recommended to refer to the implementation of coc-volar.

Reference: https://youtrack.jetbrains.com/issue/IDEABKL-7409

**Reply by weiqiyue** [reply]
Hi,does this mean that volar can be used in webstorm next? and even more plugins？
![IMG_0281](https://github.com/vuejs/language-tools/assets/63939219/9ac21c73-8e18-4e7c-bfef-e447635455b1)

**Comment by ghost** [comment, marked as answer]
From what I know, Jetbrains maintains their own, separate tooling for Vue. I've used WebStorm with Vue 2 for a while and I remember [it had features even Vetur didn't have](https://www.jetbrains.com/help/idea/vue-js.html#ws_vue_working_with_vue_components_extract_component)? 

[I see that Vue 3 related issues are fixed in the jetbrains issue tracker](https://youtrack.jetbrains.com/issue/WEB-49000) so the builtin 
Jetbrains's tools should work ok for Vue 3 (script setup etc.)

Since Jetbrains maintains their own tooling and never even used Vetur in the first place (I think), I don't think they'll support Volar officially either. If you want to use Volar anyway, you'll probably have to follow Johnson's instructions from the post above.
